spring-data-relational/src/main/java/org/springframework/data/relational/core/dialect/PostgresDialect.java

```diff
@@ -203,4 +204,21 @@ public IdentifierProcessing getIdentifierProcessing() {
 		return IdentifierProcessing.create(Quoting.ANSI, LetterCasing.LOWER_CASE);
 	}
 
+	/*
+	 * (non-Javadoc)
+	 * @see org.springframework.data.relational.core.dialect.Dialect#simpleTypes()
+	 */
+	@Override
+	public Set<Class<?>> simpleTypes() {
```

Comment:
This code could be simplified a bit to a single check by introducing a method that runs the presence check and some callback if the type is present. Check out this example in Spring Data R2DBC: https://github.com/spring-projects/spring-data-r2dbc/blob/main/src/main/java/org/springframework/data/r2dbc/dialect/PostgresDialect.java#L67